### DISCUSSION TITLE
How should user-initiated transform updates be handled on Rigidbodies?

### DISCUSSION THREAD

**Discussion by ryevdokimov** [godotengine/godot-proposals#8847] [General Discussions]
I figure this id kind of niche issue, but possibly worth discussing. 

The way things work right now are depending on what kind of hardware you are running the behavior is different. 

With a PC using a dGPU, it's a bit janky especially when fighting gravity acceleration. 

https://github.com/godotengine/godot-proposals/assets/105675984/c8a00ac6-4917-4eca-a12e-cbe6aa30c134

With a PC using an iGPU, you can't change anything at all.

![question](https://github.com/godotengine/godot-proposals/assets/105675984/6f0c7b2a-775e-403f-896b-29cdb7b2f3f1)

Neither result is particularly ideal. As well, typing in the transforms in the inspector don't seem to do anything (Only tested on my iGPU laptop)

In my opinion the way it should work is all user-input should override the physics engine as if the rigidbody were frozen and then unfrozen. 

I have a branch locally that exposes in-progress transformations when the gizmo tools are being used as vectors via EditorInterface, which allows for this by taking that vector in an editor script, checking if it's none zero, and then freezing the rigidbody. When it's zero it unfreezes, but I'm not sure if this is something people want exposed or if it's even the right solution. 

Thoughts?

**Comment by Calinou** [comment]
Making the PhysicsServer active while the editor is running isn't an actively tested use case. It lacks granularity, which severely limits its usefulness in practice. For it to be really useful in the editor, you'd need to be able to enable physics simulation on specific objects only, so you can use it to place RigidBodies in the editor without affecting the entire scene. You don't want the rest of the scene to change while you're not looking at it after all :slightly_smiling_face:

**Reply by ryevdokimov** [reply]
Personally, I've been developing an industrial simulation software that heavily uses it with great success in practice, in fact it's a critical feature 😅

Godot has been a fantastic tool to make it possible.

https://github.com/godotengine/godot-proposals/assets/105675984/91712e52-bd64-4672-9f1e-f1752ccc74e7

Interesting point about granularity, but I still think there is an underlying issue even if you were able to apply it to individual objects. My project aside, here is a hypothetical. You are developing a physics-based game, and you want to tweak several minor things in real-time to get quick feedback for rapid development. In some other engines this is somewhat trivial, however in Godot not so much, especially due to the inconsistency of intervening on physics affected rigidbodies.

**Comment by ryevdokimov** [comment]
Now that I mention it, I don't think this is specific to having physics active in the editor. I haven't used Godot how it's meant to be used much (actually building and running the project/scene), but after just playing around with it doing transforms via the remote scene in the editor also doesn't let you override the physics engine. 

I will change the title.
